Fix editor drag box visuals

peppy opened the pull request:
This has been bugging me for a while.

- Fixes dragging in a single direction not showing the box at all
- Fixes border generally looking bad due to resolution independent thickness

peppy committed:
Fix editor drag box visuals

bdach commented on osu.Game/Screens/Edit/Compose/Components/DragBox.cs:
```diff
@@ -111,5 +102,75 @@ public override void Hide()
+                    ? DrawWidth / ScreenSpaceDrawQuad.Width * 2
+                    : DrawHeight / ScreenSpaceDrawQuad.Height * 2;
+
+                Padding = new MarginPadding(-lineThickness);
```
You've mentioned on stream that you would expect the borders to "fold onto" each other when a selection has zero width or height. This can be achieved by setting the padding to half of line thickness:

```suggestion
                Padding = new MarginPadding(-lineThickness / 2);
```

Before:

https://user-images.githubusercontent.com/20418176/167127603-c9a3ae0b-726f-4c3f-a374-1374371ae270.mp4

After:

https://user-images.githubusercontent.com/20418176/167127528-1e6186d0-02da-4670-88bd-4c6568b58fab.mp4

peppy replied:
looks better, feel free to apply the patch

bdach committed:
Halve drag box padding

Allows the drag box borders to collapse in on themselves to a single
line if the drag selection has zero width or height.

bdach committed:
Merge branch 'master' into editor-drag-box
